thorlcr/master/thgraphmanager.cpp

```diff
@@ -1463,63 +1464,146 @@ void thorMain(ILogMsgHandler *logHandler, const char *wuid, const char *graphNam
+                                WARNLOG("Discarding job with invalid sessionID: wuid=%s, graph=%s (sessionID=%" I64F "d)", currentWuid.str(), currentGraphName.str(), agentSessionID);
+                                currentWuid.clear();
                             }
-                            SocketEndpoint dummyAgentEp;
-                            jobManager->execute(workunit, currentWuid, currentGraphName, dummyAgentEp);
-
-                            Owned<IWorkUnit> w = &workunit->lock();
-                            if (!multiJobLinger && lingerPeriod)
-                                w->setDebugValue(instance, "1", true);
-
-                            if (jobManager->queryExitException())
+                            else if (querySessionManager().sessionStopped(agentSessionID, 0))
                             {
-                                // NB: exitException has already been relayed.
-                                jobManager->clearExitException();
+                                WARNLOG("Discarding agentless job: wuid=%s, graph=%s", currentWuid.str(), currentGraphName.str());
+                                currentWuid.clear();
                             }
                             else
                             {
-                                switch (w->getState())
+                                SCMStringBuffer jobVersion;
+                                workunit->getDebugValue("platformVersion", jobVersion);
+                                bool mismatch = false;
+                                const char *imagePattern = getenv("imagePattern");
+                                const char *imageVersion = nullptr;
+                                if (isEmptyString(imagePattern))
+                                {
+                                    IWARNLOG("imagePattern not set in environment");
+                                    imageVersion = "unknown";
+                                    runtimePlatformVersion.set(imageVersion);
+                                }
+                                else
+                                {
+                                    imageVersion = strrchr(imagePattern, ':');
+                                    if (!imageVersion)
+                                    {
+                                        IWARNLOG("imagePattern has unexpected format: %s", imagePattern);
+                                        imageVersion = "unknown";
+                                        runtimePlatformVersion.set(imageVersion);
+                                    }
+                                    else
+                                    {
+                                        imageVersion++;
+                                        if (jobVersion.length())
+                                        {
+                                            if (0 == runtimePlatformVersion.length()) // 1st time
```

Comment:
Add a comment.
//The imagePattern corresponds to the helm chart version, but the actual image being run has been changed to runtimePlatformVersion

but this could also move outside the loop and be simplified if imagePattern was also patched in the job.